dds/DCPS/transport/framework/TransportClient.cpp

```diff
@@ -136,6 +137,8 @@ TransportClient::enable_transport_using_config(bool reliable, bool durable,
   }
   passive_connect_duration_ = TimeDuration::from_msec(duration);
 
+  populate_connection_info();
```

Comment:
This is significantly different from what was there before.  It's going to get all possible impl objects, not using `check_transport_qos`.

Reply:
I added a check to `check_transport_qos` in `populate_connection_info`.